src/components/HorizontalBarChart/index.tsx

```diff
@@ -0,0 +1,91 @@
+import { Stack, styled, Tooltip, Typography } from '@mui/material';
+import { token } from '../../utils/token';
+import { useTranslation } from '../../hooks/useTranslations';
+import { useMemo } from 'react';
+
+export interface HorizontalBarChartProps {
+	items: Array<{ color: string; title: string; count: number }>;
+}
+
+const Bar = styled('div')({
+	width: '100%',
+	height: 24,
+	display: 'flex',
+	borderRadius: token('shape.borderRadius.sm'),
+	overflow: 'hidden',
+	gap: 2,
+});
+
+const BarItem = styled('span')({
+	minWidth: 3,
+});
+
+const Legend = styled('ul')({
+	display: 'inline',
+	listStyle: 'none',
+	padding: 0,
+	margin: 0,
+});
+
+const LegendItem = styled('li')({
+	display: 'inline-flex',
+	marginRight: token('spacing.2'),
+	alignItems: 'center',
+	'&:before': {
+		content: '""',
+		backgroundColor: 'currentColor',
+		borderRadius: '50%',
+		width: '10px',
+		height: '10px',
+		display: 'inline-block',
+		marginRight: token('spacing.1'),
+	},
+});
+
+export const HorizontalBarChart = ({ items }: HorizontalBarChartProps) => {
+	const { t } = useTranslation();
+	const totalItems = useMemo(
+		() => items.reduce((sum, item) => sum + item.count, 0),
+		[items],
+	);
+	const getTotalDevicesLabel = (count: number) =>
+		count > 1
+			? t('labels.total_devices_other', { count })
+			: t('labels.total_devices_one', { count });
+
+	return (
+		<Stack sx={{ gap: token('spacing.2') }}>
+			<Bar>
+				{items
+					.filter((item) => item.count > 0)
+					.map((item) => (
+						<Tooltip
+							key={item.title}
+							title={`${item.title}: ${getTotalDevicesLabel(item.count)}`}
+						>
+							<BarItem
+								sx={{
+									backgroundColor: item.color,
+									width: `${(item.count / totalItems) * 100}%`,
+								}}
+							></BarItem>
```

Comment:
? or am I missing something?
```suggestion
						>
							<BarItem
								sx={{
									backgroundColor: item.color,
									width: `${(item.count / totalItems) * 100}%`,
								}}
							/>
```